libs/language-server/src/lib/ast/model-util.ts

```diff
@@ -187,3 +194,32 @@ export function getNextAstNodeContainer<T extends AstNode>(
   }
   return getNextAstNodeContainer(node.$container, guard);
 }
+
+/**
+ * Utility function that gets all builtin blocktypes.
+ * Make sure to call @see initializeWorkspace first so that the file system is initialized.
+ */
+export function getAllBuiltinBlocktypes(
+  documentService: LangiumDocuments,
+): BlockMetaInformation[] {
+  const allBuiltinBlocktypes: BlockMetaInformation[] = [];
+
+  documentService.all
+    .map((document) => document.parseResult.value)
+    .forEach((parsedDocument) => {
+      if (!isJayveeModel(parsedDocument)) {
+        throw new Error('Expected parsed document to be a JayveeModel');
+      }
+      parsedDocument.blocktypes.forEach((blocktypeDefinition) => {
+        if (!isBuiltinBlocktypeDefinition(blocktypeDefinition)) {
+          return;
+        }
+        if (BlockMetaInformation.canBeWrapped(blocktypeDefinition)) {
+          allBuiltinBlocktypes.push(
```

Comment:
TODO: duplicate blocktype definition

Reply:
Addressed in [7d98ae4](https://github.com/jvalue/jayvee/pull/445/commits/7d98ae4f02dafb0e4f342236e3708f8c0114095c)